django/db/migrations/state.py

```diff
@@ -352,7 +413,10 @@ def update_model_field_relation(
         remote_model_key = resolve_relation(model, *model_key)
         if remote_model_key[0] not in self.real_apps and remote_model_key in concretes:
             remote_model_key = concretes[remote_model_key]
-        self.relations[remote_model_key][model_key].append((field_name, field))
+        if field_name in self.models[model_key].fields:
+            self.relations[remote_model_key][model_key].append((field_name, field))
+        else:
+            self.relations[remote_model_key][model_key].remove((field_name, field))
```

Comment:
It looks like `update_model_field_relation()` is only called when `self._relations` is not `None`. Can you use `self._relations` here, then, instead? That would eliminate the uncertainty when reading of what code path is executing when the `relations` property is called.